packages/preact/test/index.test.tsx

```diff
@@ -131,6 +131,93 @@ describe("@preact/signals", () => {
 			await sleep();
 			expect(spy).not.to.have.been.called;
 		});
+
+		it("should support rendering JSX in Text positions", async () => {
+			const sig = signal(<span>test</span>);
+			function App({ x }: { x: typeof sig }) {
+				return <span>{x}</span>;
+			}
+
+			render(<App x={sig} />, scratch);
+
+			const text = scratch.firstChild!.firstChild!;
+
+			expect(text.textContent).to.equal("test");
+			expect(text).to.be.an.instanceOf(HTMLSpanElement);
+			expect(text).to.have.property("firstChild").that.is.an.instanceOf(Text);
+		});
+
+		it("JSX in Text should be reactive", async () => {
+			const sig = signal(<span>test</span>);
+			const spy = sinon.spy();
+			function App({ x }: { x: typeof sig }) {
+				spy();
+				return <span>{x}</span>;
+			}
+
+			render(<App x={sig} />, scratch);
+			expect(spy).to.have.been.calledOnce;
+			spy.resetHistory();
+
+			const text = scratch.firstChild!.firstChild!;
+
+			expect(text.textContent).to.equal("test");
+			expect(text).to.be.an.instanceOf(HTMLSpanElement);
+			expect(text).to.have.property("firstChild").that.is.an.instanceOf(Text);
+
+			sig.value = <div>a</div>;
+
+			expect(spy).not.to.have.been.calledOnce;
+
+			rerender();
+			scratch.firstChild!.firstChild!.textContent!.should.equal("a");
+		});
+
+		it("should support swapping between JSX and string in Text positions", async () => {
+			const sig = signal<JSX.Element | string>(<span>test</span>);
+			function App({ x }: { x: typeof sig }) {
+				return <span>{x}</span>;
+			}
+
+			render(<App x={sig} />, scratch);
+
+			{
+				const text = scratch.firstChild!.firstChild!;
+
+				expect(text.textContent).to.equal("test");
+				expect(text).to.be.an.instanceOf(HTMLSpanElement);
+				expect(text).to.have.property("firstChild").that.is.an.instanceOf(Text);
+			}
+			sig.value = "a";
+			rerender();
+			{
```

Comment:
Why the random curlies? 😅

Reply:
Because I declaring text in other block too

Reply:
`let text = ` can do that as well

Reply:
I know, but i dislike mutations without performance benefits). Will, replace with `let`)